config/karma.saucelabs.js

```diff
@@ -0,0 +1,153 @@
+ * integration.
+ */
+function getTestFiles() {
+  let root = path.resolve(__dirname, '..');
+  configs = glob.sync('{packages,integration}/*/karma.conf.js', {
+    cwd: root,
+    ignore: excluded
+  });
+  files = configs.map(x => {
+    let patterns = require(path.join(root, x)).files;
+    let dirname = path.dirname(x);
+    return patterns.map(p => path.join(dirname, p));
+  });
+  return [].concat(...files);
+}
+
+function seleniumLauncher(browserName, platform, version) {
+  return {
+    base: 'SauceLabs',
+    browserName: browserName,
+    extendedDebugging: 'true',
+    recordLogs: 'true',
+    recordVideo: 'true',
+    recordScreenshots: 'true',
+    platform: platform,
+    version: version
+  };
+}
+
+function appiumLauncher(
+  browserName,
+  deviceName,
+  platformName,
+  platformVersion
+) {
+  return {
+    base: 'SauceLabs',
+    browserName: browserName,
+    deviceName: deviceName,
+    deviceOrientation: 'portrait',
+    extendedDebugging: 'true',
+    recordLogs: 'true',
+    recordVideo: 'true',
+    recordScreenshots: 'true',
+    platformName: platformName,
+    platformVersion: platformVersion,
+    idleTimeout: 120
+  };
+}
+
+/**
+ * Custom SauceLabs Launchers
+ */
+const sauceLabsBrowsers = {
+  // Desktop
+  Chrome_Windows: seleniumLauncher('chrome', 'Windows 10', 'latest'),
+  Firefox_Windows: seleniumLauncher('firefox', 'Windows 10', 'latest'),
+  Safari_macOS: seleniumLauncher('safari', 'macOS 10.13', 'latest'),
+  Edge_Windows: seleniumLauncher('MicrosoftEdge', 'Windows 10', 'latest')
+  // IE_Windows: seleniumLauncher('internet explorer', 'Windows 10', 'latest'),
+
+  // Mobile
+  // Safari_iOS: appiumLauncher('Safari', 'iPhone Simulator', 'iOS', '11.2'),
+  // Chrome_Android: appiumLauncher('Chrome', 'Android Emulator', 'Android', '6.0')
+};
+
+module.exports = function(config) {
+  const karmaConfig = Object.assign({}, karmaBase, {
+    basePath: '../',
+
+    files: getTestFiles(),
+
+    client: Object.assign({}, karmaBase.client, {
+      firestoreSettings: {
+        host: 'firestore.googleapis.com',
+        ssl: true,
+        timestampsInSnapshots: true
+      }
+    }),
+
+    logLevel: config.LOG_INFO,
+
+    preprocessors: { '**/test/**/*.ts': ['webpack', 'sourcemap'] },
+
+    frameworks: ['mocha'],
+
+    browserNoActivityTimeout: 900000,
+
+    browserDisconnectTolerance: 3,
+
+    captureTimeout: 120000,
+
+    // browsers: ['ChromeHeadless'],
+    browsers: Object.keys(sauceLabsBrowsers),
+
+    customLaunchers: sauceLabsBrowsers,
+
+    reporters: ['spec', 'saucelabs'],
+
+    port: 9876,
+
+    retryLimit: 3,
+
+    // concurrency: 10,
+
+    sauceLabs: {
+      tunnelIdentifier: process.env.TRAVIS_JOB_NUMBER,
+      username: process.env.SAUCE_USERNAME,
+      accessKey: process.env.SAUCE_ACCESS_KEY,
+      startConnect: true
+    }
+  });
+
+  /* https://docs.travis-ci.com/user/pull-requests/#Pull-Requests-and-Security-Restrictions */
+  if (process.env.TRAVIS_PULL_REQUEST != 'false') {
```

Comment:
Per comment above: 

Let's go ahead and run these tests only during the push builds and preserve the resource if we can (see: https://docs.travis-ci.com/user/conditional-builds-stages-jobs/)

We can probably remove this block.

Reply:
Done.